drivers: wifi/airoc: Remove dependency on Infineon SOC family selection

npal-cy opened the pull request:
Issue:
Kconfig WIFI_AIROC should not depend on infineon SOC family selection for the connection. See https://github.com/zephyrproject-rtos/zephyr/issues/77012 for details.

Fix:
- update modules/hal_infineon/Kconfig to be able enable USE_INFINEON_ABSTRACTION_RTOS for non Infineon SOC family.

- fix CONFIG_NET_STATISTICS_ETHERNET to CONFIG_NET_STATISTICS_WIFI in airoc_wifi.c

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
modules: hal_infineon: Remove dependency on IFX SOC for ABSTR_RTOS

Issue:
Kconfig WIFI_AIROC should not depend on infineon SOC family selection
for the connection. See issue#77012 for details.

Fix:
- update modules/hal_infineon/Kconfig to be able enable
USE_INFINEON_ABSTRACTION_RTOS for non Infineon SOC family.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

jukkar commented on drivers/wifi/infineon/airoc_wifi.c:
```diff
@@ -761,7 +761,7 @@ static const struct wifi_mgmt_ops airoc_wifi_mgmt = {
 	.ap_enable = airoc_mgmt_ap_enable,
 	.ap_disable = airoc_mgmt_ap_disable,
-#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+#if defined(CONFIG_NET_STATISTICS_WIFI)
```
Could you split this change to different commit as the changes are not related to each other at all.

npal-cy replied:
done

npal-cy committed:
drivers: wifi/airoc: Update condition to enable NET_STATISTICS

Updated condition to enable NET_STATISTICS. Changed
CONFIG_NET_STATISTICS_ETHERNET to CONFIG_NET_STATISTICS_WIFI
in airoc_wifi.c

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

ivanwagner commented:
How many approvals does a PR such as this need to be merged?

jhedberg commented:
> How many approvals does a PR such as this need to be merged?

The assignee + 1.

You can follow which PRs are considered merge-ready here: https://merge-list.zephyrproject.io

The official (and detailed) definition of merge criteria can be found here: https://docs.zephyrproject.org/latest/project/project_roles.html#merge-criteria
